lib/posix-process/signal/sigset.c

```diff
@@ -1,55 +1,56 @@
-/* taken from newlib */
-#include "sigset.h"
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
 
 #include <errno.h>
 #include <signal.h>
-#include <uk/syscall.h>
 
-#if UK_LIBC_SYSCALLS
-int sigemptyset(sigset_t *set)
-{
-	uk_sigemptyset(set);
-	return 0;
-}
+#include <uk/syscall.h>
 
-int sigfillset(sigset_t *set)
-{
-	uk_sigfillset(set);
-	return 0;
-}
+#include "sigset.h"
 
+#if UK_LIBC_SYSCALLS
```

Comment:
I am not sure `UK_LIBC_SYSCALLS` was the correct check in the beginning either. I don't think these are syscalls right? I would use `!HAVE_LIBC`, either as a guard here or in the Makefile.

Reply:
Ooops good catch thanks. Pushed an update. Will keep the conditional into the C file for now as in `Makefile.uk` it's already conditionally compiled to `CONFIG_LIBPOSIX_PROCESS_SIGNAL`. Once we move signals out of `libposix-process` we can move the conditional into `Makefile.uk`.